src/ModuleBlacklist/Magerun/SecurityScanCommand.php

```diff
@@ -28,70 +33,199 @@ protected function configure()
+                    : ''
+            ),
+            OutputInterface::VERBOSITY_QUIET
+        );
+
+        $output->writeln(sprintf('<comment>Installed:</comment>  %s', $row->getVersion()));
+        $output->writeln(sprintf('<comment>Fixed In:</comment>   %s', $row->getFixedIn() ?: '(unknown)'));
+
+        if (!empty($row->getUpdateUrl())) {
+            $output->writeln(sprintf('<comment>Update URL:</comment> %s', $row->getUpdateUrl()));
+        }
+
+        if (!empty($row->getCredit())) {
+            $output->writeln(sprintf('<comment>Credit:</comment>     %s', $row->getCredit()));
+        }
+
+        $output->writeln('');
+
+        return true;
+    }
+
+    /**
+     * Check for match in frontend routes if module is unknown
+     *
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     * @param \Varien_Object $row
+     * @return bool
+     */
+    protected function checkIsInstalledRoute(OutputInterface $output, \Varien_Object $row)
+    {
+        $module = $this->getModuleByRoute($row->getFrontname());
+
+        if ($module === null || ($row->getName() !== '?' && !empty($row->getName()))) {
+            return false;
         }
+
+        $output->writeln(
+            sprintf(
+                '<error>Potential vulnerable module found: %s%s</error>',
+                $module,
+                $output->isQuiet() ? sprintf(' (route match: %s)', $row->getFrontname()) : ''
+            ),
+            OutputInterface::VERBOSITY_QUIET
+        );
+
+        $output->writeln(
+            '<info>'
+            . 'Matched by route: This may be a false positive where your installed module' . "\n"
+            . 'shares it with a vulnerable module, but it should be investigated further.' . "\n"
+            . 'Please contribute info about the module to MageVulnDb if it is relevant.'
+            . '</info>'
+        );
+
+        $output->writeln(sprintf('<comment>Route:</comment>      %s', $row->getFrontname()));
+        $output->writeln(sprintf('<comment>Looks Like:</comment> %s', $row->getRoute()));
+        $output->writeln(sprintf('<comment>Module:</comment>     %s', $module));
+        $output->writeln(sprintf('<comment>Installed:</comment>  %s', $this->getModuleVersion($module)));
+
+        if (!empty($row->getUpdateUrl())) {
+            $output->writeln(sprintf('<comment>Update URL:</comment> %s', $row->getUpdateUrl()));
+        }
+
+        if (!empty($row->getCredit())) {
+            $output->writeln(sprintf('<comment>Credit:</comment>     %s', $row->getCredit()));
+        }
+
+        $output->writeln('');
+
+        return true;
+    }
+
+    /**
+     * Get the module tag of the installed module matching the given route (if any).
+     *
+     * @param string $frontName
+     * @return string|null
+     */
+    protected function getModuleByRoute($frontName)
+    {
+        if ($this->routeMap === null) {
+            $routers = \Mage::getConfig()->getNode()->frontend->routers->asArray();
```

Comment:
Hmm, I think we should be concerned about admin routers too. Attackers escalate admin access to RCE taking advantage of POI vulnerabilities in admin routers.

Reply:
Given that most of the time there's no frontname for us to latch onto for admin routes, any thoughts on how we might do that?

Reply:
Good question. I guess we'd actually need to check the filenames (or class names?) to identify. I think it's not worth blocking what you've done on this and that could be handled in a later phase.